pkg/credentialwatcher/secretwatcher.go

```diff
@@ -139,20 +139,24 @@ func (s *secretWatcher) updateAccountRotateCredentialsStatus(log logr.Logger, ac
 		return
 	}
 
-	if accountInstance.Status.RotateCredentials != true {
-
-		if credentialType == "console" {
-			accountInstance.Status.RotateConsoleCredentials = true
-			log.Info(fmt.Sprintf("AWS console credentials secret was created ago requeueing to be refreshed"))
-		} else if credentialType == "cli" {
-			accountInstance.Status.RotateCredentials = true
-			log.Info(fmt.Sprintf("AWS cli credentials secret was created ago requeueing to be refreshed"))
-		}
+	if (accountInstance.Status.RotateConsoleCredentials && credentialType == "console") || (accountInstance.Status.RotateCredentials && credentialType == "cli") {
+		// We don't want to update the status since its already been updated
+		return
+	}
 
-		err = s.UpdateAccount(accountInstance)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("Error updating account %s", accountName))
-		}
+	if !accountInstance.Status.RotateConsoleCredentials && credentialType == "console" {
+		accountInstance.Status.RotateConsoleCredentials = true
+		log.Info(fmt.Sprintf("AWS console credentials secret was created ago requeueing to be refreshed"))
+	}
+
+	if !accountInstance.Status.RotateCredentials && credentialType == "cli"  {
```

Comment:
Oh, you have a gofmt fail here.
```suggestion
	if !accountInstance.Status.RotateCredentials && credentialType == "cli" {
```